### PR TITLE
Update drawing.cpp

### PR DESCRIPTION
**Pull request by zcx-xxx**
optimize FillEdgeCollection function

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/imgproc/src/drawing.cpp [hunk 1]
```diff
@@ -1457,16 +1459,16 @@ FillEdgeCollection( Mat& img, std::vector<PolyEdge>& edges, const void* color )
                     te->next = last;
                     prelast = te;
                     sort_flag = 1;
+                    last_exchange = prelast;
```
Comment (alalek) [inline modules/imgproc/src/drawing.cpp:1462]: Perhaps we don't need `sort_flag` anymore.

`sort_flag == 1` iff `last_exchange != NULL`

---

```
if (last_exchange == NULL)
    break;
```

before "while"

Comment (zcx-xxx) [reply modules/imgproc/src/drawing.cpp:1462]: Yes, that's right

Comment (alalek) [reply modules/imgproc/src/drawing.cpp:1462]: @zcx-xxx Could you please to update PR?

Comment (zcx-xxx) [reply modules/imgproc/src/drawing.cpp:1459]: Sorry, I haven't read it recently. The PR has been updated

**Review by alalek — APPROVED** [review]
Thank you :+1:

**Pull request merged**